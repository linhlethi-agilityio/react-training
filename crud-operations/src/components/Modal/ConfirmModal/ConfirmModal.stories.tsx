import type { Meta, StoryObj } from '@storybook/react';

// Components
import ConfirmModal from '.';

const meta: Meta<typeof ConfirmModal> = {
  component: ConfirmModal,
};

export default meta;
type Story = StoryObj<typeof ConfirmModal>;

export const Default: Story = {
  args: {
    isOpen: true,
    description: 'Are you sure you would like to delete student',
    title: 'Delete Student',
    buttonLabel: 'Submit',
    onConfirm: () => null,
    onCancel: () => null,
  },
};
