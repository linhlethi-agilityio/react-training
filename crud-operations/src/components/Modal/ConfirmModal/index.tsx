import { memo } from 'react';
import { Box, Button, Flex, Text } from '@chakra-ui/react';

// Components
import { CustomModal } from '@components';

interface ConfirmModalProps {
  isOpen: boolean;
  description: string;
  buttonLabel: string;
  onConfirm?: () => void;
  onCancel: () => void;
  title: string;
}

const ConfirmModal = ({ isOpen, onCancel, title, onConfirm, description, buttonLabel }: ConfirmModalProps) => {
  return (
    <CustomModal isOpen={isOpen} onClose={onCancel} title={title}>
      <Box paddingBottom={30}>
        <Text textAlign="center">{description}</Text>
        <Flex textAlign="center" gap={5} justifyContent="space-between" marginTop={5}>
          <Button onClick={onConfirm} flex={1}>
            {buttonLabel}
          </Button>
          <Button onClick={onCancel} variant="outline" flex={1}>
            CanCel
          </Button>
        </Flex>
      </Box>
    </CustomModal>
  );
};

export default memo(ConfirmModal);
