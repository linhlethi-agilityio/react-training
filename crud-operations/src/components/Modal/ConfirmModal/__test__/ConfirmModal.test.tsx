import { render } from '@test-utils';
import '@testing-library/jest-dom';

// Components
import ConfirmModal from '..';

const mockProps = {
  isOpen: true,
  description: 'Are you sure you would like to delete student',
  title: 'Delete Student',
  buttonLabel: 'Submit',
  onConfirm: jest.fn(),
  onCancel: jest.fn(),
};

describe('ConfirmModal component', () => {
  test('should match snapshot for ConfirmModal', () => {
    const { container } = render(<ConfirmModal {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});
