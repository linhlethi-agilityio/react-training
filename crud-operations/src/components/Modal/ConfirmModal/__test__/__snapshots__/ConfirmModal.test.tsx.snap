// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal component should match snapshot for ConfirmModal 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;
